renku/core/commands/providers/olos.py

```diff
@@ -0,0 +1,273 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""OLOS API integration."""
+
+import datetime
+import os
+import urllib
+from pathlib import Path
+from urllib import parse as urlparse
+from uuid import UUID, uuid4
+
+import attr
+import requests
+from tqdm import tqdm
+
+from renku.core import errors
+from renku.core.commands.providers.api import ExporterApi, ProviderApi
+from renku.core.utils import communication
+from renku.core.utils.requests import retry
+
+OLOS_SANDBOX_URL = "https://sandbox.dlcm.ch/"
+
+
+@attr.s
+class OLOSProvider(ProviderApi):
+    """Provider for OLOS integration."""
+
+    _server_url = attr.ib(default=None)
+
+    @staticmethod
+    def supports(uri):
+        """Check if provider supports a given uri for importing."""
+        return False
+
+    @staticmethod
+    def supports_export():
+        """Whether this provider supports dataset export."""
+        return True
+
+    @staticmethod
+    def export_parameters():
+        """Returns parameters that can be set for export."""
+        return {
+            "dlcm-server": ("DLCM server base url.", str),
+        }
+
+    def find_record(self, uri, client=None):
+        """Find record by uri."""
+        return None
+
+    def get_exporter(self, dataset, access_token):
+        """Create export manager for given dataset."""
+        return OLOSExporter(dataset=dataset, access_token=access_token, server_url=self._server_url)
+
+    def set_parameters(self, client, *, dlcm_server=None, **kwargs):
+        """Set and validate required parameters for a provider."""
+        CONFIG_BASE_URL = "server_url"
```

Comment:
Please use lower case variable to make the linter silent: `Variable in function should be lowercase`.